File: 0003/problem3.py
import sys
sys.setrecursionlimit(10000)

def factor(x, f):
	if x % f == 0:
		print("factor "+str(f))
		return factor(x/f, f)
		
	if (f<x):
		return factor(x, f+1)
	else:
		print("largest prime factor: {}".format(f))

def problem3(n):
	factor(n, 2)	
	
problem3(600851475143)

File: Sources/main/Problems/problem3.swift
func factor(x: Int, f: Int) {
    if (x % f == 0) {
        print("factor \(f)")
        return factor(x: x/f, f: f)
    }
    
    if (f<x) {
        return factor(x: x, f: f+1)
    } else {
        print("largest prime factor: \(f)")
    }
}

func problem3(n: Int) {
    factor(x: n, f: 2)
}
